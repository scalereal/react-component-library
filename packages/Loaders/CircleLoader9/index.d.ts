import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const CircleLoader9: React.FC<Props> & WithStyle;
export default CircleLoader9;
