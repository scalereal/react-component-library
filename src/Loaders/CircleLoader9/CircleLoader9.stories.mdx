import CircleLoader9 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/CircleLoader9"
    component={CircleLoader9}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <CircleLoader9 {...args} />
    </div>
);

<Preview>
    <Story name="CircleLoader9" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <CircleLoader9 size="S" Color="green" />
        <CircleLoader9 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <CircleLoader9 size="XS" />
        <CircleLoader9 size="S" />
        <CircleLoader9 size="M" />
        <CircleLoader9 size="L" />
        <CircleLoader9 size="XL" />
    </div>
</Preview>

### Props

<Props story="CircleLoader9" />
