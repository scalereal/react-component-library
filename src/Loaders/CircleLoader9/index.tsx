import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const CircleLoader9:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

CircleLoader9.displayName="CircleLoader9";
CircleLoader9.defaultProps = {
    size:'S',
    display: true
};

export default CircleLoader9;
