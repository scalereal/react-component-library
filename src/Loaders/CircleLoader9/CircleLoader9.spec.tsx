import { render } from '@testing-library/react';
import * as React from 'react';
import CircleLoader9 from '.';

describe('CircleLoader9 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<     CircleLoader9/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <     CircleLoader9
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <     CircleLoader9 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});