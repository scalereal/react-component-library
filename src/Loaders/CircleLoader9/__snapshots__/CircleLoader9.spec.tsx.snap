// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLoader9 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz jVwPke"
  >
    <div
      class="sc-bdfBwQ dqCOSp"
    />
  </div>
</div>
`;

exports[`CircleLoader9 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz itwfOQ"
  >
    <div
      class="sc-bdfBwQ kDXqDg"
    />
  </div>
</div>
`;

exports[`CircleLoader9 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz FEXmB"
  >
    <div
      class="sc-bdfBwQ dqCOSp"
    />
  </div>
</div>
`;
